lib/issuer.js

```diff
@@ -245,6 +244,42 @@ api.claim = function(actor, id, callback) {
   });
 };
 
+function getCredentialSigningKey(credential, callback) {
+  async.auto({
+    getIssuer: function(callback) {
+      var issuer = credential.issuer;
+      brIdentity.get(null, issuer, function(err, results) {
+        callback(err, results);
+      });
+    },
+    getPreferredSigningKey: ['getIssuer', function(callback, results) {
+      var identity = results.getIssuer;
+      var keyId = null;
+
+      if(identity.sysPreferences && identity.sysPreferences.credentialSigningKey) {
+        keyId = identity.sysPreferences.credentialSigningKey;
+      }
+      if(!keyId) {
+        return callback();
+      }
+      brKey.getPublicKey({id: keyId}, identity,
+        function(err, publicKey, meta, privateKey) {
+        publicKey.privateKey = privateKey;
+        callback(err, publicKey);
+      });
+    }]
+  }, function(err, results) {
```

Comment:
Not sure about the error handling here. All errors above will be discarded and below error used instead. May want to at least pass along `err` as the final cause argument.  That would help in debugging if `brIdentity.get` or `brKey.getPublicKey` resulted in an error.